packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1299,6 +1309,16 @@ function handleError(root, thrownValue) {
         stopProfilerTimerIfRunningAndRecordDelta(workInProgress, true);
       }
 
+      // TODO: I think it's fine to use instanceof here because this exception
+      // is always thrown by the renderer? Or use a brand check.
+      if (thrownValue instanceof SuspendOnTask) {
```

Comment:
@sebmarkbage I assume you won't like this but I legit don't know what the best approach is. A brand check seems unnecessary since this is always thrown by the renderer and not across realms or packages.